site/app/libraries/DiffViewer.php

```diff
@@ -530,21 +530,34 @@ private function getDisplay($lines, $type = self::EXPECTED, $option = self::SPEC
         $start = null;
         $html = "<div class='diff-container'><div class='diff-code'>\n";
 
+        $num_blanks = 0;
         if (isset($this->add[$type]) && count($this->add[$type]) > 0) {
             if (array_key_exists(-1, $this->add[$type])) {
+                $num_blanks = $this->add[$type][-1];
                 $html .= "\t<div class='highlight' id='{$this->id}{$type}_{$this->link[$type][-1]}'>\n";
-                for ($k = 0; $k < $this->add[$type][-1]; $k++) {
+                for ($k = 0; $k < $num_blanks; $k++) {
                     $html .= "\t<div class='row bad'><div class='empty_line'>&nbsp;</div></div>\n";
                 }
                 $html .= "\t</div>\n";
             }
         }
+        if (count($lines) + $num_blanks > 1000) {
+            if ($type === self::EXPECTED) {
+                $html = "<p style='color: black;'>This file has been truncated. Please contact instructor if you feel that you need the full file.</p>" . $html;
+            }
+            elseif ($type === self::ACTUAL) {
+                $html = "<p style='color: black;'>This file has been truncated. Please download it to see the full file.</p>" . $html;
```

Comment:
Why append `$html` here? Isn't this just going to be a thousand lines of blank content?

Reply:
The diff viewer will only show the first 1000 lines. Once it hits that amount then it will still show 1000 lines but also the message above to let you know what happened. The content won't necessarily be blank but it could if that's what the program filled STDOUT with.